gemrb/core/Game.cpp

```diff
@@ -1995,25 +1995,35 @@ bool Game::TimeStoppedFor(const Actor* target)
 	return true;
 }
 
-//recalculate the party's infravision state
-void Game::Infravision()
+/*
+ * Infravision: on specific actor with infravision or whole party with
+ *  at least one infravision actor and group-wide setting enabled.
+ */
```

Comment:
Are you saying that in the original, if you didn't select a pc with infravision, it didn't matter if anyone else had it?

Reply:
Yes. For BG2 I just verified: Without "Group infravision" being ticked, it's reserved for exlusive selection of IV PCs. When ticked, we only need at least one such a guy in the selection.

Reply:
You overwrite it globally when passed an actor though.

Reply:
To my understanding, this rule then has no effect at all.

Reply:
I mean that you change `hasInfra` unconditionally; the setting is currently ignored. So if the ini option is on and you select a char without IV, it will disable it even if another PC has it. Am I reading it wrong?

Reply:
If the INI option is on and you select a (single one or set of) char without IV, it will disable it even if another PC has it. Yes.

That's how BG2 actually behaves.

Reply:
That just sounds broken, I'm for overriding that behaviour.

Reply:
This is the description I picked from my German BG2 `dialog.tlk` - it's the same for BG, IWD2 just calls it "darkvision":

> Aktiviert diese Option, wenn Infravision für alle Charaktere angezeigt werden soll, auch wenn nur einer der ausgewählten Charaktere über Infravision verfügt. Ist diese Option deaktiviert, wird Infravision nur angezeigt, wenn tatsächlich ein Charakter mit Infravision ausgewählt ist.

Which translates to:

> Enable this option to apply infravision for every character even if only just one selected character has infravision. If this option is disaled, infravision is only used if a character is selected that actually has infravision.

To me this sounds intentional.

If not, what kind of override would you prefer?

Reply:
Enable this option to apply infravision for every character even if only just one ~selected~ character has infravision.

Reply:
Ok, did that. Even with settings disabled, the originals flawed a little when selecting e.g. twi IV PCs by shutting down the light again. This doesn't.